tools/godotcpp.py

```diff
@@ -390,7 +395,135 @@ def make_doc_source(target, source, env):
     g.close()
 
 
+def convert_size(size_bytes: int) -> str:
+    import math
+
+    if size_bytes == 0:
+        return "0 bytes"
+    SIZE_NAMES = ["bytes", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB"]
+    index = math.floor(math.log(size_bytes, 1024))
+    power = math.pow(1024, index)
+    size = round(size_bytes / power, 2)
+    return f"{size} {SIZE_NAMES[index]}"
+
+
+def get_size(start_path: str = ".") -> int:
+    total_size = 0
+    for dirpath, _, filenames in os.walk(start_path):
+        for file in filenames:
+            path = os.path.join(dirpath, file)
+            total_size += os.path.getsize(path)
+    return total_size
+
+
+def clean_cache(cache_path: str, cache_limit: int, verbose: bool):
+    from glob import glob
+
+    files = glob(os.path.join(cache_path, "*", "*"))
+    if not files:
+        return
+
+    # Remove all text files, store binary files in list of (filename, size, atime).
+    purge = []
+    texts = []
+    stats = []
+    for file in files:
+        try:
+            # Save file stats to rewrite after modifying.
+            tmp_stat = os.stat(file)
+            # Failing a utf-8 decode is the easiest way to determine if a file is binary.
+            try:
+                with open(file, encoding="utf-8") as out:
+                    out.read(1024)
+            except UnicodeDecodeError:
+                stats.append((file, *tmp_stat[6:8]))
+                # Restore file stats after reading.
+                os.utime(file, (tmp_stat[7], tmp_stat[8]))
+            else:
+                texts.append(file)
+        except OSError:
+            print(f'Failed to access cache file "{file}"; skipping.')
+
+    if texts:
+        count = len(texts)
+        for file in texts:
+            try:
+                os.remove(file)
```

Comment:
I feel a bit uncomfortable to have a build script clean out an arbitrary folder. If we have a bug in the path somehow, it would be a disaster. Though I'm not sure how best to "sanity check" it since a path _could_ be anywhere. Maybe SCons knows?

Reply:
It *shouldn't* ever be incorrect, but you make a good point. I'll see if there's a way to get a validated path from SCons directly

Reply:
Integrated! Now the path passed is explicitly the one resolved by SCons

Reply:
I had a bit of a search too and I found this: https://github.com/SCons/scons/blob/8e216dae1c61f4303117e59cba7bf1f3a7d1049c/SCons/Script/Main.py#L402-L404
Perhaps it is sufficient to iterate files in `SCons.Environment.CleanTargets`?

Reply:
That looks to be the code for `--clean` operations, and I'm not certain if that can safely overlap with a standard build. I'll look more into that at some later point, but I'm keeping this implementation for now